browser/ui/webui/settings/brave_origin/brave_origin_handler.cc

```diff
@@ -0,0 +1,308 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/ui/webui/settings/brave_origin/brave_origin_handler.h"
+
+#include <memory>
+
+#include "base/check.h"
+#include "base/command_line.h"
+#include "base/containers/fixed_flat_map.h"
+#include "base/files/file_path.h"
+#include "base/path_service.h"
+#include "base/process/launch.h"
+#include "base/task/thread_pool.h"
+#include "brave/components/ai_chat/core/common/pref_names.h"
+#include "brave/components/brave_news/common/pref_names.h"
+#include "brave/components/brave_origin/brave_origin_state.h"
+#include "brave/components/brave_rewards/core/pref_names.h"
+#include "brave/components/brave_wallet/common/pref_names.h"
+#include "brave/components/brave_wayback_machine/buildflags/buildflags.h"
+#include "brave/components/constants/pref_names.h"
+#include "brave/components/p3a/pref_names.h"
+#include "brave/components/speedreader/common/buildflags/buildflags.h"
+#include "chrome/browser/browser_process.h"
+#include "chrome/browser/profiles/profile.h"
+#include "components/metrics/metrics_pref_names.h"
+#include "components/prefs/pref_service.h"
+
+#if BUILDFLAG(ENABLE_BRAVE_WAYBACK_MACHINE)
+#include "brave/components/brave_wayback_machine/pref_names.h"
+#endif
+
+#if BUILDFLAG(ENABLE_SPEEDREADER)
+#include "brave/components/speedreader/speedreader_pref_names.h"
+#endif
+
+#if BUILDFLAG(ENABLE_BRAVE_VPN)
+#include "brave/components/brave_vpn/common/pref_names.h"
+#endif
+
+#if BUILDFLAG(ENABLE_TOR)
+#include "brave/components/tor/pref_names.h"
+#endif
+
+namespace {
+
+// Configuration for profile preferences
+struct ProfilePrefConfig {
+  const char* pref_name;
+  bool inverted;
```

Comment:
Can you explain a bit why `inverted` is necessary?

Reply:
Added a comment explaining this in https://github.com/brave/brave-core/pull/30360/commits/53f9ecb46e50b4eff887129278868ca345a5b4f3